.github/workflows/job-precommit-sql.yml

```diff
@@ -0,0 +1,86 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# To learn more about GitHub Actions in Apache Beam check the CI.md
+
+# Runs Runs Java PreCommit SQL scheduled on cron, on push to master and on pull request
+
+name: PreCommit SQL
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['sdks/java/extensions/sql**']
+permissions: read-all
+
+jobs:
+  set-properties:
+    runs-on: [self-hosted, ubuntu-20.04]
+    outputs:
+      properties: ${{ steps.test-properties.outputs.properties }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          persist-credentials: false
+          submodules: false
+      - id: test-properties
+        uses: ./.github/actions/setup-default-test-properties
+
+  sql-precommit:
+    needs: set-properties
+    name: Run PreCommit SQL
+    runs-on: [self-hosted, ubuntu-20.04]
+    strategy:
+      fail-fast: false
+      matrix:
+        java: ${{fromJson(needs.set-properties.outputs.properties).JavaTestProperties.SUPPORTED_VERSIONS}}
+        exclude: #Skipping Java 11 and 17 due to issues caused by new access options added in JDK 9 and onwards.
```

Comment:
Did this issue exist on Jenkins/why is it showing up on GHA now?

Also, could you file an issue for it?

Reply:
Oh, about this and similar ones: in jenkins, the jar files are being compiled with JDK8 and executed with JDK11 or 17. Most of this errors crop up when compiling with JDK11/17. We tried going the same route in GHA with no success as of yet. We decided to leave the corresponding jenkins jobs untouched.

I opened this issue https://github.com/apache/beam/issues/23068 to resolve this.

Reply:
> We tried going the same route in GHA with no success as of yet

What blockers are you running into? I think we should probably try to resolve the issue before merging this and other PRs (and definitely before calling the migration complete)

Relatedly, for comments like this, could you link to the issue inline in the comment as well? That both makes it easier to find all instances of the issue when we fix the problem and keeps all the context in one place.

Reply:
For JDK11, the issue is that starting with JDK9 some ByteBuffer methods changed visibility, resulting in NoSuchMethodError exceptions at compile time.

As for the blocker we're facing, we tried adding compile-time flags in Gradle (--add-exports and --add-opens) to circumvent that change in JDK9, but we kept getting the same compile-time error.

We consulted this with Kerry, and at that moment we decided to keep the associated jenkins jobs and exclude that matrix from the GHA workflows.